nntrainer/include/tensor.h

```diff
@@ -41,6 +41,7 @@ extern "C" {
 
 namespace nntrainer {
 
+class DeferredTensor;
```

Comment:
I think this looks like cyclic dependency. How about implement chain and run inside tensor and split math operation into separate file something like tensor_op.h?